packages/slack/src/post.ts

```diff
@@ -0,0 +1,202 @@
+		await client.chat.postEphemeral({
+			text: helpCmd(flags.cmd, flags.target, 'slack'),
+			user: user_id,
+			channel: channel_id,
+			thread_ts,
+		});
+		return;
+	}
+
+	if (flags.cmd === 'auth') {
+		switch (flags.target) {
+			case AuthSubcommand.Login:
+				await authLogin(client, payload);
+				return;
+			case AuthSubcommand.Logout:
+				await authLogout(client, payload);
+				return;
+			case AuthSubcommand.Status:
+				await authStatus(client, payload);
+				return;
+			default:
+				await client.chat.postEphemeral({
+					text: helpCmd(flags.cmd, flags.target, 'slack'),
+					user: user_id,
+					channel: channel_id,
+					thread_ts,
+				});
+				return;
+		}
+	}
+
+	const postMeasurements = buildPostMeasurements(flags);
+
+	// We post measurement first to catch any validation errors before committing to processing request message
+	logger.debug(`Posting measurement: ${JSON.stringify(postMeasurements)}`);
+	let measurements: PostMeasurementResponse[];
+
+	const localUserId = getLocalUserId(payload);
+	const token = await oauth.GetToken(localUserId);
+	try {
+		measurements = await postMeasurement(postMeasurements, token || undefined);
+	} catch (error) {
+		let e = error;
+		if (
+			error instanceof PostError &&
+			(error.response.statusCode === 401 ||
+				error.response.statusCode === 403) &&
+			token
+		) {
+			const errMsg = await oauth.TryToRefreshToken(localUserId, token);
+			if (errMsg) {
+				e = new Error(errMsg);
+			}
+		}
+		throw e;
+	}
+	await client.chat.postEphemeral({
+		text: '```Processing request...```',
+		user: user_id,
+		channel: channel_id,
+		thread_ts,
+	});
+	logger.debug(`Post response: ${JSON.stringify(measurements)}`);
+	logger.debug(`Latency mode: ${flags.latency}`);
+
+	let first = true;
+	// You can have multiple locations run in parallel
+	for (const measurement of measurements) {
+		const res = await getMeasurement(measurement.id);
+		logger.debug(`Get response: ${JSON.stringify(res)}`);
+		// Only want this to run on first measurement
+		if (first) {
+			await client.chat.postMessage({
+				channel: channel_id,
+				thread_ts,
+				text: `<@${user_id}>, here are the results for \`${cmdText}\``,
+			});
+			first = false;
+		}
+
+		await measurementsChatResponse(
+			logger,
+			client,
+			channel_id,
+			thread_ts,
+			measurement.id,
+			res,
+			flags
+		);
+	}
+};
+
+async function authLogin(client: WebClient, payload: ChannelPayload) {
+	const userInfoRes = await client.users.info({ user: payload.user_id });
+	const { user } = userInfoRes;
+	if (!user) {
+		logger.error('Failed to get user info');
+		await client.chat.postEphemeral({
+			text: 'Failed to get user information',
+			user: payload.user_id,
+			channel: payload.channel_id,
+			thread_ts: payload.thread_ts,
+		});
+		return;
+	}
+	const canAuthenticate = user.is_admin || user.is_owner;
+	if (!canAuthenticate) {
+		await client.chat.postEphemeral({
+			text: 'You do not have permission to authenticate',
```

Comment:
Let's update this to `Only workspace owners or administrators can use this command.` and apply it to all auth commands.